### DISCUSSION TITLE
Paste/enter bug with small amount of segments

### DISCUSSION THREAD

**Discussion by johnascroft** [JanDeDobbeleer/oh-my-posh#4255] [Troubleshoot]
### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### What happened?

When using oh-my-posh (latest version) with git bash (v2.42.0), within Microsoft Terminal (latest version) I'm getting weird behaviour when pasting commands in and pressing enter. It seems to repeat the first few characters upon pressing enter:

![image](https://github.com/JanDeDobbeleer/oh-my-posh/assets/101874136/7f44a0f8-c151-41d9-a47c-2f618aac25c8)

If I add some more segments to the config (for example adding the shell type segment), I don't seem to get the issue:

![image](https://github.com/JanDeDobbeleer/oh-my-posh/assets/101874136/78aa1633-f0bd-46d5-9c79-f24451501463)

In both cases, the commands run as expected but it seems with a minimum length of segments it seems to have this weird behaviour.

### Theme

```json
{
  "$schema": "https://raw.githubusercontent.com/JanDeDobbeleer/oh-my-posh/main/themes/schema.json",
  "blocks": [
    {
      "alignment": "left",
      "type": "prompt",
      "segments": [
        {
          "type": "shell",
          "style": "powerline",
          "powerline_symbol": "",
          "foreground": "#D8DEE9",
          "background": "#3B4252",
          "properties": {
            "mapped_shell_names": {
              "pwsh": "PS"
            }
          }
        },
        {
          "type": "path",
          "background": "#3B4252",
          "foreground": "#D8DEE9",
          "properties": {
            "style": "folder"
          },
          "style": "powerline",
          "powerline_symbol": "",
          "template": " \ue5ff {{ .Path }} "
        },
        {
          "type": "git",
          "background": "#81A0C1",
          "foreground": "#3B4252",
          "powerline_symbol": "",
          "properties": {
            "cherry_pick_icon": "\u2713 ",
            "commit_icon": "\u25b7 ",
            "fetch_status": true,
            "merge_icon": "\u25f4 ",
            "no_commits_icon": "[no commits]",
            "fetch_upstream_icon": true,
            "rebase_icon": "\u2c62 ",
            "tag_icon": "\u25b6 "
          },
          "style": "powerline",
          "template": "{{ if gt .Ahead 0 }} \u2191{{.Ahead}}{{end}}{{ if gt .Behind 0 }} \u2193{{.Behind}}{{end}}{{ if .Working.Changed }} +{{ .Working.Added }} ~{{ .Working.Modified }} -{{ .Working.Deleted }} {{ end }}"
        },
        {
          "type": "root",
          "background": "#ffc107",
          "foreground": "#18354c",
          "powerline_symbol": "",
          "style": "powerline",
          "template": " \uf0e7 "
        }
      ]
    }
  ],
  "final_space": true,
  "version": 2
}
```

### What OS are you seeing the problem on?

Windows

### Which shell are you using?

bash

### Log output

```Shell
Version: 18.5.0

Shell: bash (5.2.15(1)-release)

Prompt:

 bash   code 

Segments:

ConsoleTitle(false)                       -   0 ms
Shell(true)                               -   1 ms
Path(true)                                -   2 ms
Git(false)                                -   2 ms
Root(false)                               -   0 ms

Run duration: 9.3302ms

Cache path: C:\Users\john\AppData\Local\oh-my-posh

Config path: C:/Users/john/john.omp.json
```

**Comment by JanDeDobbeleer** [comment]
@crowngasjohn I can't reproduce this with your config, does this also happen with only oh-my-posh in your `.bashrc`?

**Comment by johnascroft** [comment]
Apologies, this is the "broken" config that causes the problem (if I remove enough segments, in this case the shell one although I don't think it's specific to that particular type):

```json
{
  "$schema": "https://raw.githubusercontent.com/JanDeDobbeleer/oh-my-posh/main/themes/schema.json",
  "blocks": [
    {
      "alignment": "left",
      "type": "prompt",
      "segments": [
        {
          "type": "path",
          "background": "#3B4252",
          "foreground": "#D8DEE9",
          "properties": {
            "style": "folder"
          },
          "style": "powerline",
          "powerline_symbol": "",
          "template": " \ue5ff {{ .Path }} "
        },
        {
          "type": "git",
          "background": "#81A0C1",
          "foreground": "#3B4252",
          "powerline_symbol": "",
          "properties": {
            "cherry_pick_icon": "\u2713 ",
            "commit_icon": "\u25b7 ",
            "fetch_status": true,
            "merge_icon": "\u25f4 ",
            "no_commits_icon": "[no commits]",
            "fetch_upstream_icon": true,
            "rebase_icon": "\u2c62 ",
            "tag_icon": "\u25b6 "
          },
          "style": "powerline",
          "template": "{{ if gt .Ahead 0 }} \u2191{{.Ahead}}{{end}}{{ if gt .Behind 0 }} \u2193{{.Behind}}{{end}}{{ if .Working.Changed }} +{{ .Working.Added }} ~{{ .Working.Modified }} -{{ .Working.Deleted }} {{ end }}"
        },
        {
          "type": "root",
          "background": "#ffc107",
          "foreground": "#18354c",
          "powerline_symbol": "",
          "style": "powerline",
          "template": " \uf0e7 "
        }
      ]
    }
  ],
  "final_space": true,
  "version": 2
}
```

If I remove everything from my `.bashrc` file and just leave this, it's the same issue:

```bash
# Initialise oh-my-posh on start-up with my own config file
eval "$(oh-my-posh init bash --config ~/john.omp.json)"
```

**Comment by JanDeDobbeleer** [comment]
@crowngasjohn same, can't reproduce it. Something else is interfering here locally. Do you have shell integrations enabled in Windows Terminal perhaps? I'm also using the Atlas engine (and preview Terminal on latest).

**Comment by chunderbolt** [comment]
I got this too - long story short, for me it was because Bash wasn't running in a login shell. That may also be the case for you too, @johnascroft.

Here's a comparison of running `C:/Program Files/Git/bin/bash.exe` with no args vs with `-i`, `-l`, or `-i -l`:

![bash compare](https://github.com/JanDeDobbeleer/oh-my-posh/assets/59027738/6a16c0cf-588e-4d0b-a9d6-850a064cd4ed)

The easiest fix for this is just ensuring your `commandline` variable in your Git Bash profile includes `-l` or `--login`. Not sure if it's worth an FAQ entry, @JanDeDobbeleer?

I was writing up more of my testing on this before I figured out the easiest solution, so I may as well post some of it.

- I don't have shell integrations enabled (turned them on to check)
- Windows Terminal Version: 1.18.3181.0 (and 1.19.231113002-preview)
- Atlas renderer on, but it happens regardless of whether it's enabled or not

Here's the cut-down theme I used in the screenshots:
  
```json
{
  "$schema": "https://raw.githubusercontent.com/JanDeDobbeleer/oh-my-posh/main/themes/schema.json",
  "blocks": [
    {
      "alignment": "left",
      "newline": true,
      "segments": [
        {
          "foreground": "#22da6e",
          "style": "plain",
          "template": "\ue285\ue285",
          // "template": ">",
          "type": "text"
        }
      ],
      "type": "prompt"
    }
  ],
  "final_space": true,
  "version": 2
}
```

If you use that in a non-login shell (as login shells don't exhibit the behaviour) you'll find that the alternative `template` of `>` doesn't cause the issue. It might be related to [the already-known issue with WT rendering certain glyphs](https://ohmyposh.dev/docs/faq#windows-terminal-unexpected-space-between-segmentstext), but I've no idea how that interacts with the lack of a login shell.
